api/v1alpha1/float64.go

```diff
@@ -0,0 +1,81 @@
+package v1alpha1
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+)
+
+//Float64 is a work around to allow representing floating points as strings
+// +kubebuilder:validation:Pattern="^(?:[-+]?(?:[0-9]+))?(?:\\.[0-9]*)?(?:[eE][\\+\\-]?(?:[0-9]+))?$"
```

Comment:
Am i right? This kubebuilder validation pattern doesn't seem to match the pattern generated in the CRDs or the tests?

```
pattern: ^0\.\d+|1\.0$ 
```

Reply:
This is for a full `float64` value, `pattern: ^0\.\d+|1\.0$ ` is just for matching `Percent` . A `Percent` is still a valid `float64`

Reply:
Ok - Can we have some tests for this scary looking regexp? or if we copied and pasted it from somewhere, what do you think if we provide link to the src so we can track it?

Reply:
done c680f77

Added link to the source https://github.com/asaskevich/govalidator/blob/7a23bdc65eef5f3783e782b436f3065eae3fc72d/patterns.go#L19

Tests are included there, it is a go validation library.

Reply:
awesome